### PR TITLE
Updated registration

### PR DESCRIPTION
**Pull request by samnotfound404**

### CODE REVIEW

File: src/helpers/api.ts [hunk 1]
```diff
@@ -203,6 +203,35 @@ export const fetchStudentData = async (
...
+    filter ? url(`/registrations?${filter}`) : url("/registrations"),
+    {
+      next: { tags: ["AllStudents"] },
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      },
+    }
+  );
+  const json = await res.json();
+  console.log(json);
```
Comment (mittal-ishaan) [inline src/helpers/api.ts:224]: remove

File: src/components/NewTableComponent/StudentModal.tsx [hunk 2]
```diff
@@ -160,19 +256,19 @@ export default function StudentModal({ open, setOpen, id }) {
...
                                                     </TableCell>
-                                                    <TableCell>{studentData.user.name}</TableCell>
+                                                    <TableCell>{studentData.user ? studentData.user.name : 'N/A'}</TableCell>
                                                 </TableRow>
                                                 <TableRow>
                                                     <TableCell component="th" scope="row" sx={{ fontWeight: 'bold' }}>
                                                         Email
                                                     </TableCell>
-                                                    <TableCell>{studentData.user.email}</TableCell>
+                                                    <TableCell>{studentData.user ? studentData.user.email : 'NA'}</TableCell>
```
Comment (mittal-ishaan) [inline src/components/NewTableComponent/StudentModal.tsx:265]:
```suggestion
                                                    <TableCell>{studentData.user ? studentData.user.email : 'N/A'}</TableCell>
```
change it everywhere

File: src/components/NewTableComponent/StudentModal.tsx [hunk 3]
```diff
@@ -49,32 +53,124 @@ const style = {
...
+            });
+
+            if (!response.ok) {
+                throw new Error(`Error fetching registration data: ${response.statusText}`);
+            }
+
+            const allData = await response.json();
+            const filteredData = allData.filter((registration) => registration.student.id === studentId);
+
+            console.log('Filtered registration data:', filteredData);
```
Comment (mittal-ishaan) [inline src/components/NewTableComponent/StudentModal.tsx:143]: remove

File: src/components/NewTableComponent/StudentModal.tsx [hunk 4]
```diff
@@ -49,32 +53,124 @@ const style = {
...
+            setRegistrationData(filteredData);
+        } catch (error) {
+            console.error('Error fetching registration data:', error);
```
Comment (mittal-ishaan) [inline src/components/NewTableComponent/StudentModal.tsx:146]: remove

File: src/helpers/api.ts [hunk 5]
```diff
@@ -203,6 +203,34 @@ export const fetchStudentData = async (
...
+      next: { tags: ["AllStudents"] },
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      },
+    }
+  );
+  const json = await res.json();
+  return json;
+};
+
```
Comment (mittal-ishaan) [inline src/helpers/api.ts:226]: remove these empty lines

File: src/helpers/api.ts [hunk 6]
```diff
@@ -203,6 +203,34 @@ export const fetchStudentData = async (
...
+  const res = await fetch(
+    filter ? url(`/registrations?${filter}`) : url("/registrations"),
+    {
+      next: { tags: ["AllStudents"] },
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      },
+    }
+  );
+  const json = await res.json();
```
Comment (mittal-ishaan) [inline src/helpers/api.ts:223]:
```suggestion
  return await res.json();
```

File: src/components/NewTableComponent/StudentModal.tsx [hunk 7]
```diff
@@ -49,32 +53,123 @@ const style = {
...
+          
+        } finally {
+            setLoading(false);
+        }
+    };
 
+    useEffect(() => {
         if (open && id) {
-            fetchStudentData();
+            fetchStudentData(Cookies.get("accessToken"), id);
```
Comment (mittal-ishaan) [inline src/components/NewTableComponent/StudentModal.tsx:153]: lets fetch accessToken only once here and use in both of them.

File: src/components/NewTableComponent/StudentModal.tsx [hunk 8]
```diff
@@ -49,32 +53,123 @@ const style = {
...
+        setLoading(true);
+        try {
+            const response = await fetch(`${baseUrl}/api/v1/registrations`, {
+                headers: {
+                    Authorization: `Bearer ${accessToken}`
+                }
+            });
+
+            if (!response.ok) {
+                throw new Error(`Error fetching registration data: ${response.statusText}`);
```
Comment (mittal-ishaan) [inline src/components/NewTableComponent/StudentModal.tsx:136]: what will this throw do? comment on the behaviour in this case

File: src/helpers/api.ts (hunk 6 above)
Comment (samnotfound404) [reply src/helpers/api.ts:223]: couldn't change it due to some error in promises

**Review by mittal-ishaan — APPROVED** [review]

**Pull request merged**